File: 61-rotate-list/rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not head or not head.next or k == 0:
            return head

        temp=head
        size=1

        while(temp.next):
            temp=temp.next
            size+=1

        k=k%size

        if k==0:
            return head

        loop=size-k
        tail=head

        while(loop-1):
            tail=tail.next
            loop-=1
            
        curr_head=tail.next
        tail.next=None
        temp.next=head
        return curr_head
       






        

File: 61-rotate-list/rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if (head==NULL || k==0 || head->next==NULL) return head;
        ListNode * temp=head;
        int size=1;
        while(temp->next!=NULL){
            temp=temp->next;
            size+=1;
        }
        k%=size;
        if(k==0)return head;
        ListNode * tail=head;
        int loop=size-k-1;
        while(loop>0){
            tail=tail->next;
            loop-=1;
        }
        ListNode * new_head=tail->next;
        tail->next=NULL;
        temp->next=head;
        return new_head;
    }
};